### PR TITLE
fix(ident/models): Allow trait value to support multiple type

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

**Comment by gagantrivedi** [comment]
@matthewelwell  not sure if this should be a major release?

**Comment by matthewelwell** [comment]
We need some tests for this please. For now, I suggest that we either: 

1. Write a test to load / dump data using the entire TraitSchema
2. Just write a unit test to test the `serialize` and `deserialize` methods on the `TraitSchema` class.

**Comment by matthewelwell** [comment]
@gagantrivedi I think we could get away with just putting this in a v1.2.1. It's just a bug fix, really. But I'm equally happy with v1.3.0 really so we can just leave it as is.

File: tests/identities/test_trait_schema.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "trait_data",
+    (
+        {"trait_key": "key", "trait_value": "value"},
+        {"trait_key": "key1", "trait_value": 21},
+        {"trait_key": "key1", "trait_value": None},
+        {"trait_key": "key1", "trait_value": 11.2},
+        {"trait_key": "key1", "trait_value": True},
+    ),
+)
+def test_trait_schema_load_and_dump(trait_data):
```
Comment (matthewelwell) [inline tests/identities/test_trait_schema.py:16]: We should split this into 2 separate tests and split up the parametrized inputs to make the tests more readable: 

```python
@pytest.mark.parametrize(
    "trait_key, trait_value",
    (
         ("key", "value"),
         ("key", 21), 
         ...
    )
)
def test_trait_schema_load(trait_key, trait_value):
    # Given
    trait_schema = TraitSchema()

    # When
    trait_model = trait_schema.load({"trait_key": trait_key, "trait_value": trait_value})

    # Then
    assert trait_model.trait_key == trait_key
    assert trait_model.trait_value == trait_value


@pytest.mark.parametrize(
    "trait_model",
    (
         TraitModel("key", "value"),
         TraitModel("key", 21), 
         ...
    )
)
def test_trait_schema_load(trait_model):
    # Given
    trait_schema = TraitSchema()

    # When
    trait_data = trait_schema.dump(trait_model)

    # Then
    assert trait_data["trait_key"] == trait_model.trait_key
    assert trait_data["trait_value"] == trait_model.trait_value
```

**Review by matthewelwell — CHANGES_REQUESTED** [review]

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**